client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -225,36 +228,41 @@ protected static final void setApplicationProperties(final Message msg, final Ma
     }
 
     /**
-     * Starts a timer with the delay configured in {@link ClientConfigProperties#getInactiveLinkTimeout()}. The timer
-     * checks if a message has been sent on the sender link in this period (subclass needs to call
-     * {@link #storeLastSendTime()} when sending a message). If no message has been sent in the timout period, it closes
-     * the links, otherwise a new timer is scheduled to recheck.
+     * Starts a timer with the delay configured by {@link ClientConfigProperties#getInactiveLinkTimeout()} and
+     * {@link ClientConfigProperties#getInactiveLinkTimeoutUnit()}. The timer checks if a message has been sent on the
+     * sender link in this period (subclass needs to call {@link #storeLastSendTime()} when sending a message). If no
+     * message has been sent in the timout period, it closes the links, otherwise a new timer is scheduled to recheck.
      * <p>
      * This method needs to be called by subclasses where {@link #sender} is set.
      * <p>
-     * If {@link ClientConfigProperties#getInactiveLinkTimeout()} is <code>0</code>, this method does nothing.
+     * If {@link ClientConfigProperties#getInactiveLinkTimeout()} is &lt; 1, this method does nothing.
      */
     protected final void startAutoCloseTimer() {
-        final long inactiveLinkTimeout = connection.getConfig().getInactiveLinkTimeout();
-        if (inactiveLinkTimeout > 0) {
-            startAutoCloseTimer(inactiveLinkTimeout);
+        final long inactiveLinkTimeoutMillis = getInactiveLinkTimeoutMillis();
+        if (inactiveLinkTimeoutMillis > 0) {
+            startAutoCloseTimer(inactiveLinkTimeoutMillis);
         }
     }
 
+    private long getInactiveLinkTimeoutMillis() {
+        final TimeUnit unit = connection.getConfig().getInactiveLinkTimeoutUnit();
+        return unit.toMillis(connection.getConfig().getInactiveLinkTimeout());
+    }
+
     /**
      * Stores the current time stamp on the sender link. This is used to detect and close inactive sender links as
      * documented in {@link #startAutoCloseTimer()}.
      */
     protected final void storeLastSendTime() {
-        sender.attachments().set(ATTACHMENT_LAST_SEND_TIME, Long.class, Instant.now().toEpochMilli());
+        sender.attachments().set(KEY_LAST_SEND_TIME, Long.class, System.currentTimeMillis());
```

Comment:
the JavaDoc refers to the private *startAutoCloseTimer* method's documentation. However, that method has no JavaDoc and even if it had, you couldn't `@link` to it ...

Reply:
It links to the method `protected final void startAutoCloseTimer()`, which has a JavaDoc (the method `startAutoCloseTimer(final long delay)`, which takes a parameter is private).